app/controllers/employees/rewards/orders_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Employees
+  module Rewards
+    class OrdersController < EmployeesController
+      before_action :set_reward
+      def new
+        @order = Order.new
+      end
+
+      def create
+        # flash[:alert]= 'You do not have enough points for that reward'
+        @order = @reward.orders.new(employee: current_employee)
+        if @order.save
+          current_employee.number_of_earned_points -= @reward.price
+          @current_employee.save
```

Comment:
What happens if db will go offline between lines 12 and 14, and we will not save `@current_employee.save`??

Reply:
I've applied two transactions. Sorry for fogeting to do that in the first place.